### DISCUSSION TITLE
Bandwidth of THD+N measurement

### DISCUSSION THREAD

**Discussion by khaut** [endolith/waveform-analysis#37] [Q&A]
Hi,

What do you think about thd.py THD+N measurement bandwidth? I believe it's something that is reported many times together with THD+N measurement results. And for audio typically 20Hz - 20kHz.

If someone wants to do that .. should the high pass / low pass be done in time domain for the data before it's sent thd.py functions? Or should it be done in frequency domain with numpy inside thd.py functions after rfft and before irfft ? Maybe default to something or optionally to also work in nonstandard measurements?

Thanks :)

**Comment by khaut** [comment]
After searching a bit I think it's best to do in time domain for the data before sending to thd.py

**Comment by endolith** [comment]
Well for this script it would just be determined by the sample rate, no? 

You could add your own low pass before the script and report that, if you wanted

**Reply by khaut** [reply]
Yes,  if needed, adding low pass (and hi pass) before this script should be the way to go I believe.

Sample rate does limit but it might be 192 kHz (or higher) so in that case you get passband up to 96kHz. If you have noise above 20 kHz (quite possible) and it's included you might end up with THD+N number that does not represent 20Hz - 20 kHz in case that's what you would like to specify. According to some, frequency range should specified for THD+N to be meaningful. And in some cases you might specify some other bandwidth also. 100-10kHz to have nice marketing value :|
